### PR TITLE
Update ImGuiComponents & ImGuiHelpers

### PR DESCRIPTION
**Pull request by ItsBexy**
Took some helper functions created for `UiDebug2`, and incorporated them into `ImGuiComponents` and `ImGuiHelpers` instead

- `IconButton()` and its various overloads now include an optional size parameter
- `IconButtonSelect()` has been added, allowing a row or grid of IconButtons to serve as a radio-like input
- `HelpMarker()` now includes an optional color parameter
- `ClickToCopyText()` now includes an optional color parameter, and includes the `FontAwesome.Copy` icon in the tooltip.
- Updated methods to use ImRaii where applicable

These changes are intended not to break any existing calls in plugins or within Dalamud itself (knock on wood).

### CODE REVIEW

File: Dalamud/Interface/Components/ImGuiComponents.DisabledButton.cs [hunk 1]
```diff
@@ -21,15 +23,17 @@ public static partial class ImGuiComponents
...
+        bool button;
+        using (ImRaii.PushFont(UiBuilder.IconFont))
+        {
+            var text = icon.ToIconString();
+            if (id.HasValue)
+            {
+                text = $"{text}##{id}";
+            }
+
+            button = DisabledButton(text, defaultColor, activeColor, hoveredColor, alphaMult);
```
Comment (Infiziert90) [inline Dalamud/Interface/Components/ImGuiComponents.DisabledButton.cs:35]: You can just return here, no need to store it just for return

Return ends the scope automatically

File: Dalamud/Interface/Utility/ImGuiHelpers.cs [hunk 2]
```diff
@@ -167,17 +167,41 @@ public static List<Vector4> DefaultColorPalette(int swatchCount = 32)
...
+
+                ImGui.SameLine();
+                ImGui.TextUnformatted($"{textCopy}");
+            }
         }
 
-        if (ImGui.IsItemClicked()) ImGui.SetClipboardText($"{textCopy}");
+        if (ImGui.IsItemClicked())
+        {
+            ImGui.SetClipboardText($"{textCopy}");
```
Comment (Infiziert90) [inline Dalamud/Interface/Utility/ImGuiHelpers.cs:203]: Could just directly use `textCopy`, it gets null checked above and is a string already

**Comment by ItsBexy** [comment]
I've also put together a [GameIconButton](https://github.com/goatcorp/Dalamud/blob/dae0074d87976e00cdc9f0c3b75b9902f8c30bd6/Dalamud/Interface/Components/ImGuiComponents.GameIconButton.cs) feature, but I haven't included it in this PR because it relies on `ITextureProvider` and I'm not yet sure what would be the least clunky way to address that dependency.

File: .gitignore [hunk 3]
```diff
@@ -327,4 +327,5 @@ ASALocalRun/
 *.nvuser
 
 # MFractors (Xamarin productivity tool) working folder 
-.mfractor/
\ No newline at end of file
+.mfractor/
+/Dalamud/Interface/Components/ImGuiComponents.GameIconButton.cs
```
Comment (KazWolfe) [inline .gitignore:331]: Eh?

**Pull request closed without merging**